core/src/main/scala/org/http4s/EntityDecoder.scala

```diff
@@ -71,6 +79,18 @@ sealed trait EntityDecoder[T] { self =>
   // shamelessly stolen from IList
   def widen[B](implicit ev: T <~< B): EntityDecoder[B] =
     ev.subst[({type λ[-α] = EntityDecoder[α @uncheckedVariance] <~< EntityDecoder[B]})#λ](refl)(this)
+
+  def asLenient = new EntityDecoder[T] {
+    override def decode(msg: Message): DecodeResult[T] = this.decode(msg)
```

Comment:
You've made yourself a infinite loop here by using `this.decode`, you should be using `self.decode`.